ext/opentelemetry-ext-docker-tests/tests/redis/test_redis_functional.py

```diff
@@ -0,0 +1,143 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import redis
+
+from opentelemetry import trace
+from opentelemetry.ext.redis import RedisInstrumentor
+from opentelemetry.test.test_base import TestBase
+
+
+class TestRedisInstrument(TestBase):
+
+    test_service = "redis"
+
+    def setUp(self):
+        super().setUp()
+        self.redis_client = redis.Redis(port=6379)
+        self.redis_client.flushall()
+        RedisInstrumentor().instrument(tracer_provider=self.tracer_provider)
+
+    def tearDown(self):
+        super().tearDown()
+        RedisInstrumentor().uninstrument()
+
+    def test_long_command(self):
+        self.redis_client.mget(*range(1000))
+
+        spans = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans), 1)
+        span = spans[0]
+        self.assertEqual(span.attributes["service"], self.test_service)
+        self.assertEqual(span.name, "redis.command")
+        self.assertIs(
+            span.status.canonical_code, trace.status.StatusCanonicalCode.OK
+        )
+
+        self.assertEqual(span.attributes.get("db.instance"), 0)
+        self.assertEqual(
```

Comment:
is it worth refactoring some of these assertions into a utility method? there's a lot of identical code with regards to db.instance / url.

Reply:
good call, refactored.